### PR TITLE
BUGFIX avoid silent overflow in `lcm` least common multiple function

### PR DESCRIPTION
**Pull request by KlausC**
Solves #33911 .
Define new functions `checked_lcm` and `widelcm` analogously to `checked_mul` and `widemul`.

The behaviour of `lcm` is not changed, and the user can switch to the appropriate variant at his own discretion.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
The `lcm` function is so costly that making it checked by default seems reasonable to me. Similarly, I feel like explicitly widening before calling `lcm` would be more composable.

**Comment by KlausC** [comment]
So it would be ok to change `lcm` by using `checked_mul` internally and neither introduce `widelcm` nor `checked_lcm`.
To be honest, I think that is the best fix, but I wanted to avoid the "checking is too expensive" - discussion.

**Comment by StefanKarpinski** [comment]
It's all about context: checking is too expensive when you're doing `i + j` but when you're doing `lcm(i, j)` that's a totally different story—that's already a very expensive operation.

**Comment by KlausC** [comment]
As you can see, I retired the new functions and added the `checked_mul` to `lcm`. The `widelcm` function is probably not worth the effort. So what remains is the fix of the silent error in `lcm`.

File: test/checked.jl [hunk 1]
```diff
@@ -166,6 +166,7 @@ import Base: checked_abs, checked_neg, checked_add, checked_sub, checked_mul,
     @test checked_cld(typemin(T), T(1)) === typemin(T)
     @test_throws DivideError checked_cld(typemin(T), T(0))
     @test_throws DivideError checked_cld(typemin(T), T(-1))
+
```
Comment (StefanKarpinski) [inline test/checked.jl]: I think you can revert the changes to this file now.

**Comment by StefanKarpinski** [comment]
Well, that's certainly a much simpler fix now 😁 👍 💯

File: test/checked.jl (hunk 1 above)
Comment (KlausC) [reply test/checked.jl]: done

**Review by StefanKarpinski — APPROVED** [review]

**Comment by KlausC** [comment]
The last change fixes a gotcha in `twiceprecision.jl`, where `lcm` is expected to produce incorrect results (and not to throw an overflow exception).
A test case failed for this reason on 32-bit machines.
 No other such uses of `lcm` were found in `base` or `stdlib` sources.

**Comment by KlausC** [comment]
is the being forgotten?

**Comment by StefanKarpinski** [comment]
Thanks for the reminder!
